### PR TITLE
Detect optislang install using pyoptislang

### PR DESCRIPTION
**Pull request by iazehaf**
- Detect optiSLang install using PyOptiSLang
- Rework page.py to speed-up initialization
- Lock treeview if analysis has not started
- Refactore code

### CODE REVIEW

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/model/optislang/install.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+"""Common functions."""
+
+from pathlib import Path
+from typing import Union
+
+from ansys.optislang.core import utils
+
+
+def get_available_optislang_installations(outout_format: str = "short") -> list:
```
Comment (ansGEsquembri) [inline src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/model/optislang/install.py:11]: Wouldn't it be `output_format`?

Comment (iazehaf) [reply src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/model/optislang/install.py]: @ansGEsquembri thanks. Corrected.

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/utils/alerts.py [hunk 2]
```diff
@@ -37,18 +37,28 @@ def update_alerts(problem_setup_step: ProblemSetupStep) -> list:
...
-        solve_message, solve_color = "optiSLang simulation not started.", "warning"
-    elif problem_setup_step.optislang_solve_status == "processing":
-        solve_message, solve_color = "optiSLang simulation in progress.", "primary"
-    elif problem_setup_step.optislang_solve_status == "stopped":
-        solve_message, solve_color = "optiSLang simulation stopped.", "danger"
-    elif problem_setup_step.optislang_solve_status == "aborted":
-        solve_message, solve_color = "optiSLang simulation aborted.", "danger"
-    elif problem_setup_step.optislang_solve_status == "finished":
-        solve_message, solve_color = "optiSLang simulation completed successfully.", "success"
+    if problem_setup_step.project_state == "NOT STARTED":
```
Comment (ansGEsquembri) [inline src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/utils/alerts.py:40]: Could this long `if` be changed to a dictionary instead?

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/views/summary_page.py [hunk 3]
```diff
@@ -2,23 +2,57 @@
...
+    prevent_initial_call=True,
+)
+def update_design_table(n_intervals, pathname):
+
+    project = DashClient[{{ cookiecutter.__solution_definition_name }}].get_project(pathname)
+    problem_setup_step = project.steps.problem_setup_step
+
+    summary_view = SummaryView()
+
+    if problem_setup_step.selected_actor_from_treeview in problem_setup_step.actors_info.keys():
```
Comment (ansGEsquembri) [inline src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/views/summary_page.py:53]: Maybe this piece of code should be put in a method so there's no need to repeat it.

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/utils/alerts.py (hunk 2 above)
Comment (iazehaf) [reply src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/utils/alerts.py]: Done.

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/views/summary_page.py (hunk 3 above)
Comment (iazehaf) [reply src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/views/summary_page.py:53]: @ansGEsquembri thank you for your proposal. We plan to do a refactoring of the code soon. Let's keep that for a another PR.

**Review by ansGEsquembri — APPROVED** [review]

**Pull request merged**